agent/consul/kubernetesidp/k8s.go

```diff
@@ -0,0 +1,343 @@
+package kubernetesidp
```

Comment:
A bunch of this was cloned from the vault k8s plugin and I'm still on the fence if anything more than plain TokenReview actually is useful (like parsing the JWT and comparing the contents against what TokenReview returns).